src/states/game.js

```diff
@@ -334,23 +396,21 @@ Game.prototype =
         return this.__uuid;
     },
 
-    __asteroid_out_of_bounds : function(asteroid)
+    __object_out_of_bounds : function(object, object_array_string)
     {
-        if (asteroid.position.x > CONSTS.SCREEN_DIMENSIONS[1])
+        if (object.position.x > CONSTS.SCREEN_DIMENSIONS[0])
         {
             return;
         }
-
-        asteroid.destroy();
-        this.__non_player_objects =
-            this.__non_player_objects.filter(function(item)
+        object.destroy();
+        this[object_array_string] =
```

Comment:
I tried passing this.__stars and this.__asteroids but I couldn't get them to update globally. Maybe you can show me how it's done?!